src/components/new-safe/create/steps/StatusStep/useSafeCreationEffects.ts

```diff
@@ -19,36 +19,32 @@ const useSafeCreationEffects = ({
   const chainId = useChainId()
   const [pendingSafe, setPendingSafe] = usePendingSafe()
 
+  // Asynchronously wait for Safe creation
   useEffect(() => {
```

Comment:
I've split this effect into 4 smaller effects, each with their own set of dependencies and actions.